In [1]:
import pandas as pd

In [2]:
wine_frame = pd.read_csv('./wine-reviews/wine_data_2.csv')
wine_frame =wine_frame.drop_duplicates(subset='title', keep='first', inplace=False)

In [3]:
# Seeing what shape the data is in.
wine_frame[:5]

Unnamed: 0   country                                        description  \
0           0     Italy  Aromas include tropical fruit, broom, brimston...   
1           1  Portugal  This is ripe and fruity, a wine that is smooth...   
2           2        US  Tart and snappy, the flavors of lime flesh and...   
3           3        US  Pineapple rind, lemon pith and orange blossom ...   
4           4        US  Much like the regular bottling from 2012, this...   

                          designation  points  price           province  \
0                        Vulkà Bianco      87    NaN  Sicily & Sardinia   
1                            Avidagos      87   15.0              Douro   
2                                 NaN      87   14.0             Oregon   
3                Reserve Late Harvest      87   13.0           Michigan   
4  Vintner's Reserve Wild Child Block      87   65.0             Oregon   

              region_1           region_2         taster_name  \
0                 Etna                NaN       Kerin O’Keefe   
1                  NaN                NaN          Roger Voss   
2    Willamette Valley  Willamette Valley        Paul Gregutt   
3  Lake Michigan Shore                NaN  Alexander Peartree   
4    Willamette Valley  Willamette Valley        Paul Gregutt   

  taster_twitter_handle                                              title  \
0          @kerinokeefe                  Nicosia 2013 Vulkà Bianco  (Etna)   
1            @vossroger      Quinta dos Avidagos 2011 Avidagos Red (Douro)   
2           @paulgwine       Rainstorm 2013 Pinot Gris (Willamette Valley)   
3                   NaN  St. Julian 2013 Reserve Late Harvest Riesling ...   
4           @paulgwine   Sweet Cheeks 2012 Vintner's Reserve Wild Child...   

          variety               winery  
0     White Blend              Nicosia  
1  Portuguese Red  Quinta dos Avidagos  
2      Pinot Gris            Rainstorm  
3        Riesling           St. Julian  
4      Pinot Noir         Sweet Cheeks

In [4]:
# ****** California Cabernets ******

In [5]:
# Create a slice that only Contains Califorina Cabernet Savignon
cali_cab =wine_frame[ wine_frame.province == 'California']
cali_cab =cali_cab[cali_cab.variety =='Cabernet Sauvignon']

In [6]:
#How many Califorina Cabernet are in the Data Set
cali_cab_count=cali_cab.country.count()
cali_cab_count 

5317

In [7]:
# What is the average rating?
cali_cab.points.mean()

89.1002444987775

In [8]:
# What is the average price?
cali_cab.price.mean()

59.60343525858815

### There are a total of 5317 Californa Cabernets in our data set, with an average rating of 89.1 and and Average Price of $59.60

This is a change of the original data of: 
There are a total of 7452 Californa Cab in our data set, with an average rating of 88.49 and and Average Price of $52.39
* the change was made after removing duplicated data

In [1]:
# For every California Cab in the Data set, 
# Returns the Rating,
# returns the average price based on Rating, 
# along with the total number of wines with that rating
# and the standerd deveation based on price

cali_cab_avgs = []
for x in range (80,101):
    temp_slice =cali_cab[cali_cab.points == x]
    mean=temp_slice.price.mean()
    count=temp_slice.price.count()
    std=temp_slice.loc[:,"price"].std()
    cali_cab_avgs.append((x,mean,count,std))
    
cali_cab_avgs  

NameError: name 'cali_cab' is not defined

## Best Buys  --  Wines that are below  average price, at  each rating

In [10]:
best_buys=[]
for x in range (20):
    temp=(cali_cab[(cali_cab.points ==cali_cab_avgs[x][0]) & (cali_cab.price <=(cali_cab_avgs[x][1]-cali_cab_avgs[x][3]))][['title','price','points']].sort_values(by='price'))
    if len(temp)>0:
        price=round(cali_cab_avgs[x][1],2)
        best_buys.append('At ' + str(cali_cab_avgs[x][0]) +' points, the average price is '+ str(price))
        best_buys.append(temp)
    else:
        best_buys.append('There are no wines that Qualify at ' + str(cali_cab_avgs[x][0]) +' points')
    best_buys.append('************************************************' )
best_buys

['At 80 points, the average price is 18.48',
                                                    title  price  points
 79093  Mountain View 2005 Cabernet Sauvignon (Central...    8.0      80
 21842     Pessimist 2012 Cabernet Sauvignon (California)   10.0      80
 36195  Coyote Creek 2012 Cabernet Sauvignon (California)   10.0      80
 92448  Arrow Creek 2005 Vintner's Selection Cabernet ...   10.0      80,
 '************************************************',
 'There are no wines that Qualify at 81 points',
 '************************************************',
 'At 82 points, the average price is 24.04',
                                                 title  price  points
 28453  Backhouse 2009 Cabernet Sauvignon (California)    6.0      82,
 '************************************************',
 'There are no wines that Qualify at 83 points',
 '************************************************',
 'At 84 points, the average price is 31.17',
                                                

# ****** Washington Cabernets ******

In [11]:
wash_cab =wine_frame[ wine_frame.province == 'Washington']
wash_cab =wash_cab[wash_cab.variety =='Cabernet Sauvignon']

In [12]:
wash_cab_count=wash_cab.country.count()
wash_cab_count 

1251

In [13]:
# What is the average rating?
wash_cab.points.mean()

89.35171862509992

In [14]:
# What is the average price?
wash_cab.price.mean()

40.99919549477072

## There are a total of 1251 Washington Cabernets in our data set, with an average rating of 89.3 and and Average Price of $40.99

In [15]:
# For every Washington Cab in the Data set, 
# Returns the Rating,
# returns the average price based on Rating, 
# along with the total number of wines with that rating
# and the standered deviaion based on Price

wash_cab_avgs = []
for x in range (80,101):
    temp_slice =wash_cab[wash_cab.points == x]
    std=temp_slice.loc[:,"price"].std()
    mean=temp_slice.price.mean()
    count=temp_slice.price.count()
    wash_cab_avgs.append((x,mean,count,std))
    
wash_cab_avgs  

[(80, nan, 0, nan),
 (81, 19.0, 4, 10.03327796219494),
 (82, 23.0, 5, 8.455767262643882),
 (83, 27.272727272727273, 11, 15.99431717261421),
 (84, 19.5, 14, 9.061753948071235),
 (85, 24.246153846153845, 65, 18.155604135871148),
 (86, 25.28358208955224, 67, 11.541500834799361),
 (87, 30.66891891891892, 148, 12.740950377047543),
 (88, 34.407407407407405, 162, 16.969668064829783),
 (89, 37.083333333333336, 156, 17.032779908102185),
 (90, 39.56983240223464, 179, 16.529829538700472),
 (91, 48.927374301675975, 179, 22.603079220716534),
 (92, 54.58181818181818, 110, 18.75989369220374),
 (93, 56.05714285714286, 70, 22.4750333070545),
 (94, 66.87179487179488, 39, 27.371882891197796),
 (95, 70.4, 20, 26.138094804327263),
 (96, 81.5, 8, 25.483888467590095),
 (97, 69.33333333333333, 3, 15.044378795195678),
 (98, 130.0, 2, 7.0710678118654755),
 (99, 125.0, 1, nan),
 (100, nan, 0, nan)]

In [16]:
wash_cab_97 =wash_cab[wash_cab.points >= 97][['title','price','points']]
sorted_wash_cab_99=wash_cab_97.sort_values(by=['price'])
sorted_wash_cab_99

title  price  points
48900   Betz Family 2005 Père de Famille Cabernet Sauv...   55.0      97
86718   Betz Family 2010 Père de Famille Cabernet Sauv...   68.0      97
33841   Abeja 2007 Reserve Cabernet Sauvignon (Columbi...   85.0      97
1556    Quilceda Creek 2008 Cabernet Sauvignon (Columb...  125.0      99
45799   Quilceda Creek 2006 Cabernet Sauvignon (Columb...  125.0      98
125829  Quilceda Creek 2009 Cabernet Sauvignon (Columb...  135.0      98

## Which are the Best Buys (Washington Cabernet)?  
##### A best buy is a wine that's price is atleast one standerd devation below the mean for its rating.

In [17]:
best_buys=[]
for x in range (21):
    temp=(wash_cab[(wash_cab.points ==wash_cab_avgs[x][0]) & (wash_cab.price <=(wash_cab_avgs[x][1]-wash_cab_avgs[x][3]))][['title','price','points']].sort_values(by='price'))

    if len(temp)>0:
        best_buys.append(temp)
    else:
        best_buys.append('There are no wines that Qualify at ' + str(cali_cab_avgs[x][0]) +' points')
best_buys

['There are no wines that Qualify at 80 points',
 'There are no wines that Qualify at 81 points',
                                                     title  price  points
 101034  Whispering Tree 2013 Cabernet Sauvignon (Colum...   10.0      82,
                                                    title  price  points
 90623  StoneCap 2011 Estate Grown Cabernet Sauvignon ...   10.0      83,
                                                    title  price  points
 74631  Hogue 2014 Cabernet Sauvignon (Columbia Valley...    8.0      84
 66128  Two Ponds 2012 Cabernet Sauvignon (Columbia Va...   10.0      84,
 'There are no wines that Qualify at 85 points',
                                                     title  price  points
 51529    Pine & Post 2007 Cabernet Sauvignon (Washington)    8.0      86
 60683   Washington Hills 2013 Cabernet Sauvignon (Wash...    9.0      86
 73818   Hogue 2006 Cabernet Sauvignon (Columbia Valley...    9.0      86
 78917      Covey Run 1999 Cabernet Sauvi